### ISSUE TITLE
Change number of epochs after when resuming training. 

### ISSUE THREAD

**Issue by mazatov** [ultralytics/ultralytics#18154]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have initially started training with 200 epochs but now that it's taking too long and the model is already performing I would like to terminate earlier. So I cancelled the training and am resuming it using :

`yolo task=detect mode=train resume model="$LAST_PT_PATH" data="$TRAIN_DATA" epochs = 130` but it still is trying to train to 200 epochs. 

I also tried editing the `args.yaml ` in the train folder but it still stuck to 200. Where is it saved to train for 200 epochs and how can I edit it?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mazatov, thank you for your interest in Ultralytics 🚀! We recommend taking a look at the [Docs](https://docs.ultralytics.com) for detailed guidance and examples. You may find the sections on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage particularly useful as they explain training configurations in detail.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) that we can use to debug the issue. 

From your description, it seems like you’re attempting to adjust the number of epochs while resuming training. Please ensure:
1. You use the correct syntax for overriding the parameters when running the training command.
2. Your modifications persist in the correct locations—e.g., `args.yaml` or explicitly passed via the command line.

If this is a ❓ Question, please share the following to help us assist you better:
- The exact command you're using
- Any relevant training logs or outputs
- The content of your `args.yaml` file

To explore similar topics or get direct help from the community, consider joining the Ultralytics community:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time conversation
- [Discourse](https://community.ultralytics.com) for in-depth discussions
- [Subreddit](https://reddit.com/r/ultralytics) for topic-based sharing

## Upgrade

First, ensure you’re using the latest version of the `ultralytics` package. Issues are often resolved in newer updates, so upgrading might solve this problem:

```bash
pip install -U ultralytics
```

Make sure you're running this in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) installed.

## Environments

For running YOLO, verified environments include:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud environments**: Options like [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Local setups**: Including Docker. See our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

When this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI runs validate all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu systems on a 24-hour cycle and after every commit.

🤖 This is an automated response, but an Ultralytics engineer will follow up with you shortly to provide further assistance!

**Comment by Y-T-G** [comment]
You can't change epochs while resuming

**Comment by mazatov** [comment]
Anything I can do? The final models are much smaller in size so I'd like to get them if possible without training much longer.

**Comment by Y-T-G** [comment]
You can run this:

```python
from ultralytics.utils.torch_utils import strip_optimizer

strip_optimizer("path/to/best.pt")
```

It will reduce the size of the models.

**Comment by banu4prasad** [comment]
> You can't change epochs while resuming

we can by changing the args.yaml file

**Comment by glenn-jocher** [comment]
Not on current Ultralytics behavior—`resume=True` reloads training args from the checkpoint itself, not `args.yaml`, and `epochs` is not one of the supported resume overrides in [`BaseTrainer.check_resume()`](https://docs.ultralytics.com/reference/engine/trainer/); if you want fewer total epochs, start a new run from `last.pt` or `best.pt` without `resume=True`, or just shrink the checkpoint with [`strip_optimizer()`](https://docs.ultralytics.com/reference/utils/torch_utils/). Since this thread is stale/closed, please open a new issue if you’re seeing different behavior on a current version.
